basxconnect/core/views/person/person_details_views.py

```diff
@@ -163,30 +163,35 @@ def togglepersonstatus(request, pk: int):
 
 
 def confirm_delete_email(request, pk: int):
-    class ConfirmDeleteEmailForm(forms.Form):
-        delete_mailer_contact = django.forms.BooleanField(
-            label=_("Delete linked email subscriptions as well"),
-            required=False,
-        )
-
     email = models.Email.objects.get(id=pk)
     enable_delete_mailer_contact_checkbox = apps.is_installed(
         "basxconnect.mailer_integration"
     ) and hasattr(email, "subscription")
 
+    if enable_delete_mailer_contact_checkbox:
+
+        from basxconnect.mailer_integration.settings import MAILER
+
+        class DeleteMailerSubscriptionForm(forms.Form):
+            delete_mailer_contact = django.forms.BooleanField(
+                label=_("Delete linked %s subscription as well") % MAILER.name(),
+                required=False,
+            )
+
     if request.method == "POST":
-        form = ConfirmDeleteEmailForm(request.POST)
+        if enable_delete_mailer_contact_checkbox:
+            form = DeleteMailerSubscriptionForm(request.POST)
+        else:
+            form = forms.Form()
         if form.is_valid():
             person = email.person
             if enable_delete_mailer_contact_checkbox and form.cleaned_data.get(
                 "delete_mailer_contact"
             ):
                 try:
-                    import basxconnect.mailer_integration.settings
+                    from basxconnect.mailer_integration.settings import MAILER
 
-                    basxconnect.mailer_integration.settings.MAILER.delete_person(
-                        email.email
-                    )
+                    MAILER.delete_person(email.email)
```

Comment:
Maybe we can simplify the nesting of ifs a bit here?

Something like
```python
    # need to take the form class definition outside of the conditional

    form = DeleteMailerSubscriptionForm()
    if request.method == "POST" and enable_delete_mailer_contact_checkbox:
        form = DeleteMailerSubscriptionForm(request.POST)
        if form.is_valid():
            person = email.person
            if form.cleaned_data.get(
                "delete_mailer_contact"
            ):
                try:
                    from basxconnect.mailer_integration.settings import MAILER

                    MAILER.delete_person(email.email)
                except Exception:
                    logging.error("tried to delete person from mailchimp but failed")

            email.delete()
            person.refresh_from_db()
            person.save()
            return HttpResponseRedirect(
                reverse_model(person, "read", kwargs={"pk": person.pk})
            )
```

Not sure 100% if this works correctly, might need improvment. Just an idea to make the flow a bit clearer.

Also, in general I would avoid to add a new name to the function namespace (here the name ```DeleteMailerSubscriptionForm```) inside a conditional. Otherwise you always have to mentally check back if this name is available and add more if-guards further down.

Reply:
I tried to make it a bit clearer by having just one big branch for the case we need the additional checkbox. To me it doesn't seem very easy to simplify it further.
One issue is that I would like to use the name "Mailchimp" (which comes from the settings of the mailer module) for the user. I think a generic term like "external mailer" is just not understandable for grannies etc.